runners/core-java/src/main/java/org/apache/beam/runners/core/SimpleDoFnRunner.java

```diff
@@ -256,6 +256,8 @@ public void onTimer(
         break;
 
       case PROCESSING_TIME:
+        effectiveTimestamp = stepContext.timerInternals().currentProcessingTime();
```

Comment:
Ah, there is a general problem here. The `effectiveTimestamp` is an event time timestamp for the elements that are output from the `@OnTimer` method. The current processing time is in a different time domain that may be different in arbitrary ways.